elastic/beats
# Remove logger outside of WriteJSONKeys

## What does this PR do?

This change is removing the creation of the logger object inside of the WriteJSONKeys method. 

That method is executed for each event causing the logger to initialize objects for each event.

Additionally adding a benchmark that show the impact of the change, for a specific scenario.
```
goos: darwin
goarch: amd64
pkg: github.com/elastic/beats/v7/libbeat/common/jsontransform
cpu: Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
                 │  base.out   │             logger.out             │
                 │   sec/op    │   sec/op     vs base               │
WriteJSONKeys-16   3.202µ ± 9%   2.589µ ± 3%  -19.13% (p=0.000 n=8)

                 │   base.out   │             logger.out              │
                 │     B/op     │     B/op      vs base               │
WriteJSONKeys-16   2.820Ki ± 0%   2.344Ki ± 0%  -16.90% (p=0.000 n=8)

                 │  base.out  │            logger.out             │
                 │ allocs/op  │ allocs/op   vs base               │
WriteJSONKeys-16   23.00 ± 0%   16.00 ± 0%  -30.43% (p=0.000 n=8)

```

Signed-off-by: Alexandros, Sapranidis <alexandros@elastic.co>


## Why is it important?

Under certain conditions for Filebeat, but not only, but the logger was also consuming overall a lot of memory. That memory then had to be reclaimed by the GC, so reducing the logger memory, actually reduces the gc work.

## Checklist



~- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~

## How to test this PR locally

1) Build and trigger filebeat to load from a file and at the end capture the heap pprof. 
2) Observe the heap memory allocation for the `logp.NewLogger`
